pkgs/development/libraries/xdg-desktop-portal-shana/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, meson
+, rustc
+, git
+, ninja
+, cargo
+, xdg-desktop-portal
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
Since it is a rust package, maybe it makes sense to use `rustPlatform.buildRustPackage`? See https://nixos.org/manual/nixpkgs/stable/#rust

Reply:
Maybe, but a big part of the installation and configuration is done with meson. I tried but I wasn't able to make the meson shell hooks run from buildRustPackage.

Reply:
Anyway, I figured it out. I will update this.